src/mscorlib/src/System/String.cs

```diff
@@ -411,7 +415,61 @@ private unsafe static bool EqualsHelper(String strA, String strB)
                 return (length <= 0);
             }
         }
-        
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        private unsafe static bool OrdinalCompareSubstring(String strA, String strB, int length)
+        {
+            Contract.Requires(strA != null);
+            Contract.Requires(strB != null);
+            Contract.Requires(strA.Length <= length);
+            Contract.Requires(strB.Length <= length);
+
+            fixed (char* ap = &strA.m_firstChar) fixed (char* bp = &strB.m_firstChar)
+            {
+                char* a = ap;
+                char* b = bp;
+
+#if AMD64
+#if ALIGN_ACCESS
+    #error Needs porting for ALIGN_ACCESS (https://github.com/dotnet/corert/issues/430) 
+#endif // ALIGN_ACCESS
```

Comment:
Thank you for collecting the data. Yes, there is inconclusive noise...

My call would be to always align so that it has high propability of working great on variaty of hardware (e.g. including ARM64).

You can reduce the overhead of the alignment by replacing `if (length >= 2)` in `StartsWithOrdinalHelper` with assert. The condition will be always true because of the caller handles lenghts 0 and 1. 

The condition can be removed it in EqualsHelper as well. There is always at least int32 allocated for the string, and the empty strings where the comparison would be unnecessary overhead pretty much never get there - they are handled by the reference equality check in the caller because of they are empty string singleton in most cases.